internal/pkg/cache/add.go

```diff
@@ -154,6 +154,9 @@ func (c *Cache) cloneRemoteGitRegistry(opts *AddOpts) (string, error) {
 	// If ref is set, add query string variable
 	if !opts.IsLatest() {
 		url = fmt.Sprintf("%s?ref=%s", url, opts.Ref)
+	} else {
+		// Attempt to shallow clone the constructed url
+		url = fmt.Sprintf("%s?depth=1", url)
```

Comment:
I'd rather have the shallow clone be a a flag to the add command than handling it this way.